File: 0969-number-of-recent-calls/0969-number-of-recent-calls.py
class RecentCounter:
    
    def __init__(self):
        self.requests=deque()

    def ping(self, t: int) -> int:
        lb=t-3000
        self.requests.append(t)
        while self.requests[0] <lb:
            self.requests.popleft()

        return len(self.requests)       


# Your RecentCounter object will be instantiated and called as such:
# obj = RecentCounter()
# param_1 = obj.ping(t)

File: 0969-number-of-recent-calls/0969-number-of-recent-calls.cpp
class RecentCounter {
public:
    RecentCounter() {
        
    }
    
    int ping(int t) {
        ios_base::sync_with_stdio(false);
        cin.tie(NULL);
        cout.tie(NULL);
        int lb=t-3000;
        requests.push(t);
        while(!requests.empty() && requests.front()<lb)
            requests.pop();
        return requests.size();
    }
private:
    queue <int> requests;
};

/**
 * Your RecentCounter object will be instantiated and called as such:
 * RecentCounter* obj = new RecentCounter();
 * int param_1 = obj->ping(t);
 */